dataprofiler/labelers/data_processing.py

```diff
@@ -2079,3 +2079,113 @@ def process(self, data, labels=None, label_mapping=None, batch_size=None):
         # predictions is the argmax of the average of the cell's label votes
         results["pred"] = np.argmax(results["pred"], axis=1)
         return results
+
+
+class ColumnNameModelPostprocessor(
+    BaseDataPostprocessor, metaclass=AutoSubRegistrationMeta
+):
+    """Subclass of BaseDataPostprocessor for postprocessing regex data."""
+
+    def __init__(self, parameters):
+        """Initialize the ColumnNameModelPostProcessor class."""
+        # set parameters
+        if not parameters:
+            parameters = {}
+        parameters.setdefault("true_positive_dict", None)
+        parameters.setdefault("positive_threshold_config", None)
```

Comment:
this ultimately should mimic the other procesors where we take in paramters as individual args: e.g.

```python
def __init__(
        self,
        default_label="UNKNOWN",
        pad_label="PAD",
        flatten_separator=" ",
        use_word_level_argmax=False,
        output_format="character_argmax",
        separators=(" ", ",", ";", "'", '"', ":", "\n", "\t", "."),
        word_level_min_percent=0.75,
    ):
        """
        Initialize the CharPostprocessor class.

        :param default_label: Key for label_mapping that is the default label
        :type default_label: string (could be int, char, etc.)
        :param pad_label: Key for label_mapping that is the pad label
        :type pad_label: string (could be int, char, etc.)
        :param flatten_separator: separator used to put between flattened
            samples.
        :type flatten_separator: str
        :param use_word_level_argmax: whether to require the argmax value of
            each character in a word to determine the word's entity
        :type use_word_level_argmax: bool
        :param output_format: (character_argmax vs NER) where character_argmax
            is a list of encodings for each character in the input text and NER
            is in the dict format which specifies start,end,label for each
            entity in a sentence
        :type output_format: str
        :param separators: list of characters to use for separating words within
            the character predictions
        :type separators: tuple(str)
        :param word_level_min_percent: threshold on generating dominant
            word_level labeling
        :type word_level_min_percent: float
        """
        super().__init__(
            default_label=default_label,
            pad_label=pad_label,
            flatten_separator=flatten_separator,
            use_word_level_argmax=use_word_level_argmax,
            output_format=output_format,
            separators=separators,
            word_level_min_percent=word_level_min_percent,
        )
```

Reply:
the super should auto handle parameters where need be,
if we want `true_positive_dict` to be a dict by default, we could always do a 
```python
if true_positive_dict is None:
    true_positive_dict = {}
```

Reply:
ah good point... I was following the `model` paradigm still. I'll update this